File: DSA 1/Bubble Sort.java
import java.util.*;

// https://course.acciojob.com/idle?question=04c51a0a-2136-40da-8738-c391d6a9ef8f

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();

        int arr[] = new int[N];

        // Input of array
        for (int i = 0; i < N; i++) {
            arr[i] = sc.nextInt();
        }

        // bubble sort in descending order
        bubbleSort(N, arr);
        // printing array elements
        for (int i = 0; i < N; i++) {
            System.out.print(arr[i] + " ");
        }

    }
	public static void swap(int[]arr,int j){
		int temp = arr[j+1];
		arr[j+1] = arr[j];
		arr[j] = temp;
	}

    public static void bubbleSort(int N, int[] arr) {
        for(int i=0;i<arr.length-1;i++){
			for(int j=0;j<N-i-1;j++){
				if(arr[j]>arr[j+1]){
					swap(arr,j);
				}
			}
		}
    }
}

File: Javascript/Bubble Sort.js
// https://course.acciojob.com/idle?question=0ef618cf-606f-4c76-a4e2-659b4ccd7113

const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});

readline.question('', N => {
  readline.question('', arr => {
    arr = arr.split(' ').map(Number);
    bubbleSort(N, arr);
    console.log(arr.join(' '));
    readline.close();
  });
});

function bubbleSort(N, arr) {

    // Bubble Sort
    for(let i=0; i<N-1; i++){
        for(let j=0; j<N-i-1; j++){
            if(arr[j] > arr[j+1]){
                // Swap the Elements
                let temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
}
